packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -245,41 +260,292 @@ class FilteringTextInputFormatter extends TextInputFormatter {
   /// two "o"s replaced by a single occurrence of the replacement
   /// string) because both of the "o"s would be matched simultaneously
   /// by the pattern.
-  ///
-  /// Additionally, each segment of the string before, during, and
-  /// after the current selection in the [TextEditingValue] is handled
-  /// separately. This means that, in the case of the "Into the Woods"
-  /// example above, if the selection ended between the two "o"s in
-  /// "Woods", even if the pattern was `RegExp('o+')`, the result
-  /// would be "Int* the W**ds", since the two "o"s would be handled
-  /// in separate passes.
-  ///
-  /// See also [String.splitMapJoin], which is used to implement this
-  /// behavior in both cases.
   final String replacementString;
 
+  /// Determines how to perform formatting, update the selection and the
+  /// composing region, on each banned pattern in the input string.
+  ///
+  /// Once the [allow] parameter and the [filterPattern] parameter has determind
+  /// the regions of the banned patterns, [replacingStrategy] will be called on
+  /// each of these regions. The default [FilteringFormatterReplacingStrategy]
+  /// is [preserveSelectionAndComposingRegion], which replaces the text within
+  /// every banned region with [replacementString], and updates the composing
+  /// region and the selection accordingly so they do not fall within the
+  /// removed regions.
+  ///
+  /// The framework also provides a [preserveSelectionAndSkipComposingRegion]
+  /// strategy, which keeps the content within the current composing region
+  /// intact, and otherwise has same behavior as
+  /// [preserveSelectionAndComposingRegion]. With this strategy, a banned word
+  /// will not be immediately replaced if it's currently being composed. In
+  /// general this is more robust than [preserveSelectionAndComposingRegion], as
+  /// most IMEs only expect composing region changes from the user, not the text
+  /// editor. This replacing strategy should be used if CJK IME users experience
+  /// duplicated input in the text field. See
+  /// https://github.com/flutter/flutter/issues/79844 for a video footage of
```

Comment:
Not sure if I should link to a github issue in public docs. But I'd like to have a video footage to show when you should apply this workaround instead of having a wall of text.

Reply:
I remember you can link a youtube video in the API doc?